### PR TITLE
Feature/changing weeks

### PR DESCRIPTION
**Pull request by Tomekkk**
After mergin to master, backend publish action needed

### CODE REVIEW

File: app/src/main/java/com/macoscope/ketchuplunch/presenter/WeeksPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+class WeeksPresenter(val weeksService: WeeksService, val weeksView: WeeksView) : AnkoLogger {
+    val subscriptions: CompositeSubscription = CompositeSubscription()
+    private val REQUEST_AUTHORIZATION = 1001
+    private fun loadData() {
+        weeksView.showLoading()
+        subscriptions += loadWeeksObservable().subscribe(
+                subscriber<List<Week>>()
+                        .onNext {
+                            weeksView.showWeeks(it)
+                            weeksView.hideLoading()
```
Comment (pziomacki) [inline app/src/main/java/com/macoscope/ketchuplunch/presenter/WeeksPresenter.kt]: Create private method `showWeeks` and move there weeksView.showWeeks(it) and weeksView.hideLoading()

File: app/src/main/java/com/macoscope/ketchuplunch/view/lunch/LunchActivity.kt [hunk 2]
```diff
@@ -1,49 +1,101 @@
...
-    override fun onOptionsItemSelected(item: MenuItem): Boolean {
-        val id = item.itemId
+    private fun setupWeeksSpinner(weeksSpinner: Spinner) {
+        weeksAdapter = WeeksAdapter()
+        weeksSpinner.adapter = weeksAdapter
+        weeksSpinner.onItemSelectedListener = this
+    }
+
+    override fun onNothingSelected(parent: AdapterView<*>?) {
+        throw UnsupportedOperationException()
```
Comment (dbacinski) [inline app/src/main/java/com/macoscope/ketchuplunch/view/lunch/LunchActivity.kt]: intended? it will crash app?

Comment (pziomacki) [inline app/src/main/java/com/macoscope/ketchuplunch/view/lunch/LunchActivity.kt]: Add TODO or implement proper beahaviour.

**Comment by pziomacki** [comment]
-2 
implement proper behaviour or add some comment instead of throwing unsupportedOperationException

File: app/src/main/java/com/macoscope/ketchuplunch/view/lunch/WeekItemUI.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                ellipsize = TextUtils.TruncateAt.END
+                textSize = 18f
+                height = dip(40f)
+                gravity = Gravity.CENTER_VERTICAL
+                textColor = Color.WHITE
+                setPadding(dip(8f), 0, dip(8f), 0)
+                backgroundColor = ContextCompat.getColor(ctx, R.color.colorRed)
+            }
+        }
+
```
Comment (dbacinski) [inline app/src/main/java/com/macoscope/ketchuplunch/view/lunch/WeekItemUI.kt]: remove \n

File: app/src/main/java/com/macoscope/ketchuplunch/view/lunch/WeeksAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package com.macoscope.ketchuplunch.view.lunch
+
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.TextView
+import com.macoscope.ketchuplunch.R
+import com.macoscope.ketchuplunch.model.lunch.Week
+import org.jetbrains.anko.AnkoContext
+import java.util.*
```
Comment (dbacinski) [inline app/src/main/java/com/macoscope/ketchuplunch/view/lunch/WeeksAdapter.kt]: use direct imports, can be changed in settings

**Comment by dbacinski** [comment]
+2

**Comment by dbacinski** [comment]
+2

**Pull request merged**